akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala

```diff
@@ -0,0 +1,403 @@
+    def name = "HandlingCommands"
+
+    final override def onCommand(command: Command): Behavior[Any] = {
+      val effect = commandHandler(commandContext, state, command)
+      applyEffects(command, effect)
+    }
+    final override def onJournalResponse(response: Response): Behavior[Any] = {
+      ???
+      // should not happen, what would it reply?
+      same
+    }
+  }
+
+  object PersistingEvents extends EventsourcedRunningPhase {
+    def name = "PersistingEvents"
+    final override def onCommand(c: Command): Behavior[Any] = {
+      log.info(s"PERSISTING EVENTS, command, STASH: ${c}")
+      stash(c, same)
+    }
+    final override def onJournalResponse(response: Response): Behavior[Any] = {
+      log.info("RESPONSE == " + response)
+      response match {
+        case WriteMessagesSuccessful ⇒ same // ignore, we only act on WriteMessageSuccess
+        case WriteMessageSuccess(p, id) ⇒
+          // instanceId mismatch can happen for persistAsync and defer in case of actor restart
+          // while message is in flight, in that case we ignore the call to the handler
+          if (id == instanceId) {
+            updateLastSequenceNr(p)
+            popApplyHandler(p.payload)
+            onWriteMessageComplete()
+          } else same
+
+        case WriteMessageRejected(p, cause, id) ⇒
+          // instanceId mismatch can happen for persistAsync and defer in case of actor restart
+          // while message is in flight, in that case the handler has already been discarded
+          if (id == instanceId) {
+            updateLastSequenceNr(p)
+            onWriteMessageComplete() // FIXME no sure if we should call it here
+            onPersistRejected(cause, p.payload, p.sequenceNr)
+          } else same
+
+        case WriteMessageFailure(p, cause, id) ⇒
+          // instanceId mismatch can happen for persistAsync and defer in case of actor restart
+          // while message is in flight, in that case the handler has already been discarded
+          if (id == instanceId) {
+            onWriteMessageComplete()
+            onPersistFailureThenStop(cause, p.payload, p.sequenceNr)
+          } else same
+
+        case LoopMessageSuccess(l, id) ⇒
+          // FIXME probably we don't need it?
+
+          // instanceId mismatch can happen for persistAsync and defer in case of actor restart
+          // while message is in flight, in that case we ignore the call to the handler
+          if (id == instanceId) {
+            popApplyHandler(l)
+            same
+          } else same
+
+        case WriteMessagesSuccessful ⇒
+          writeInProgress = false
+          flushJournalBatch()
+
+        case WriteMessagesFailed(_) ⇒
+          writeInProgress = false
+          same // it will be stopped by the first WriteMessageFailure message
+
+      }
+    }
+
+    private def onWriteMessageComplete(): Behavior[Any] = {
+      tryBecomeHandlingCommands()
+      tryUnstash(context, same)
```

Comment:
This is triggered by `onJournalResponse`, which is triggered by `onMessage` so here `stopped` should work fine

Reply:
Yes, works